### PR TITLE
Migrate S1528 to eslint implementation (no-array-constructor)

### LINKED ISSUE #2113: Migrate S1528 to eslint frontend (new rule for TS)

### PR DESCRIPTION
**Pull request by guillaume-dequenne-sonarsource**
Fixes #2113

### CODE REVIEW

File: eslint-bridge/src/rules/array-constructor.ts [hunk 1]
```diff
@@ -17,17 +17,28 @@
...
+    function checkNewExpression(node: estree.Node) {
+      const newExpression = node as estree.NewExpression;
+      if (newExpression.callee.type === 'Identifier' && newExpression.callee.name === 'Array') {
+        let message = 'Use either a literal or "Array.from()" instead of the "Array" constructor.';
+        if (
+          newExpression.arguments.length === 1 &&
+          newExpression.arguments[0].type === 'Literal' &&
+          typeof newExpression.arguments[0].value === 'number'
+        ) {
+          message = 'Use "Array.from()" instead of the "Array" constructor.';
```
Comment (andrea-guarino-sonarsource) [inline eslint-bridge/src/rules/array-constructor.ts:34]: Why did you change the message? If I'm not wrong Array.from has been introduced starting from ES6, so that may be invalid for people writing code in older JS environment

Comment (guillaume-dequenne-sonarsource) [reply eslint-bridge/src/rules/array-constructor.ts:34]: From my understanding, it's the only alternative if you want to create an array with a given initial size. 
Since calling the `Array` constructor is so error-prone, we decided to keep raising an issue for those cases nevertheless. But then, the message instructing to use a literal is not valid.

**Review by andrea-guarino-sonarsource — APPROVED** [review]
